dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -480,35 +485,73 @@ Spdp::handle_participant_data(DCPS::MessageId id, const ParticipantData_t& cpdat
       return;
     }
 
-    // Must unlock when calling into part_bit() as it may call back into us
-    ACE_Reverse_Lock<ACE_Thread_Mutex> rev_lock(lock_);
+    // Checks if sequence numbers are increasing
+
+    bool ValidSeq;
+
+    // Check if subroutne call is from Sedp Security
+
+    if (currentSeq.getValue() != 0) {
```

Comment:
I think you should get rid of `ValidSeq`, make this check part of `validateSequenceNumber()`, and put `validateSequenceNumber(currentSeq, iter)` in the `if` evaluation.